### ISSUE TITLE
Crash if there are multiple services discovered

### ISSUE THREAD

**Issue by curioustechizen** [aurae/RxBonjour#5]
This caught me off-guard. Apparently, an instance of `NsdManager.ResolveListener` cannot be re-used. That is to say, for each newly found service, you need to supply a new instance of `ResolveListener` to the `resolveService()` method. If you try to reuse the listener, you get a crash with the following trace:

```
Caused by: java.lang.IllegalArgumentException: listener already in use
            at android.net.nsd.NsdManager.resolveService(NsdManager.java:613)
            at rxbonjour.internal.JBBonjourDiscovery$1$2.onNext(JBBonjourDiscovery.java:99)
```

The easiest way to solve this is to replace the offending line with:

``` java
nsdManager.resolveService(info, createResolveListener());
```

and then, instead of having a member variable `resolveListener`, you create a new one in the method:

``` java
private NsdManager.ResolveListener createResolveListener() {
    return new NsdManager.ResolveListener() {
        //Rest of the code here
    };
}
```

I would have sent this change as a PR but I am not sure if this change will impact any of the Rx stuff.

**Comment by mannodermaus** [comment]
When trying to reproduce the bug, I found that it's triggered quite unreliably, which leads me to believe that this may be a race issue in NsdManager. This is the reason I originally implemented `Backlog` (since NsdManager can only resolve one service at a time), but supposedly that isn't enough. It shouldn't be an issue to have a new instance initialized for each service, but I'll have to test this a little more thoroughly before I can say that for sure.

**Comment by curioustechizen** [comment]
You're right - I also came across some instances when the crash doesn't occur. But in my experience it does occur 8 out of 10 times. The trick is just to be broadcasting two services simultaneously. 

If it helps, here are the commands I used on Linux to advertise two services simultaneously.

```
$ avahi-publish -s "First" "_http._tcp" 12345 -v
$ avahi-publish -s "Second" "_http._tcp" 12346 -v
```

Equivalent commands for Mac OSX and Windows would be

```
$ dns-sd -R "First" _http._tcp . 12345
$ dns-sd -R "Second" _http._tcp . 12346
```

After running these commands, I connected my Android phone to the same WiFi router that my laptop is connected to. Ran the RxBonjour example (with the `RxBonjour.startDiscovery(this, input, true)` variant of course). Now I see the crash.

Devices tested:
- Moto G (Android 5.0.2)
- Micromax Canvas A1 (Android 5.1)
- Nexus 7 2012 (Android 5.1)
